pages/api/newsletter/subscribe.ts

```diff
@@ -0,0 +1,48 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+
+export default async function handler(
+    req: NextApiRequest,
+    res: NextApiResponse
+) {
+    const { email } = req.body
+
+    if (!email) {
+        return res.status(400).json({ error: 'Email is required' })
+    }
+
+    const existingSubscribers = await fetch(
+        'https://www.getrevue.co/api/v2/subscribers',
+        {
+            method: 'GET',
+            headers: {
+                Authorization: `Token ${process.env.REVUE_API_KEY}`,
+                'Content-Type': 'application/json',
+            },
+        }
+    )
+    const subscribers = await existingSubscribers.json()
+
+    if (subscribers.some((sub) => sub.email === email)) {
+        return res
+            .status(201)
+            .json({ error: '', message: `You're already subscribed! 👌🏾` })
+    }
+
+    const result = await fetch('https://www.getrevue.co/api/v2/subscribers', {
```

Comment:
I would fault Revue for this, because I checked their API and it doesn't define what happens if someone is already subscribed. My assumption though, is that this endpoint should return an error for duplicate subscribers, and instead of having to fetch all the subscribers above, you should just be able to call this and catch the error.

If I were you, I would probably try to use Postman or a similar HTTP client and test with my own email, and see if I could figure out what error they send for "duplicate" emails and then shrink this logic down a bit.

Reply:
I was thinking something similar, the Api docs could use some work😅